build-support/functions/20-build.sh

```diff
@@ -396,73 +390,48 @@ function build_consul_local {
       build_arch="${XC_ARCH}"
    fi
 
-   local use_gox=1
-   is_set "${NOGOX}" && use_gox=0
-   which gox > /dev/null || use_gox=0
-
    status_stage "==> Building Consul - OSes: ${build_os}, Architectures: ${build_arch}"
    mkdir pkg.bin.new 2> /dev/null
-   if is_set "${use_gox}"
-   then
-      status "Using gox for concurrent compilation"
-
-      CGO_ENABLED=0 gox \
-         -os="${build_os}" \
-         -arch="${build_arch}" \
-         -osarch="!darwin/arm !darwin/arm64 !freebsd/arm"  \
-         -ldflags="${GOLDFLAGS}" \
-         -parallel="${GOXPARALLEL:-"-1"}" \
-         -output "pkg.bin.new/${extra_dir}{{.OS}}_{{.Arch}}/consul" \
-         -tags="${GOTAGS}" \
-         .
-
-      if test $? -ne 0
-      then
-         err "ERROR: Failed to build Consul"
-         rm -r pkg.bin.new
-         return 1
-      fi
-   else
-      status "Building sequentially with go install"
-      for os in ${build_os}
+
+   status "Building sequentially with go install"
+   for os in ${build_os}
+   do
+      for arch in ${build_arch}
       do
-         for arch in ${build_arch}
-         do
-            outdir="pkg.bin.new/${extra_dir}${os}_${arch}"
-            osarch="${os}/${arch}"
-            if test "${osarch}" == "darwin/arm" -o "${osarch}" == "darwin/arm64" -o "${osarch}" == "freebsd/arm64" -o "${osarch}" == "windows/arm" -o "${osarch}" == "windows/arm64" -o "${osarch}" == "freebsd/arm"
-            then
-               continue
-            fi
-
-            if test "${os}" == "solaris" -a "${arch}" != "amd64"
-            then
-               continue
-            fi
-
-            echo "--->   ${osarch}"
-
-
-            mkdir -p "${outdir}"
-            GOBIN_EXTRA=""
-            if test "${os}" != "$(go env GOHOSTOS)" -o "${arch}" != "$(go env GOHOSTARCH)"
-            then
-               GOBIN_EXTRA="${os}_${arch}/"
-            fi
-            binname="consul"
-            if [ $os == "windows" ];then
-                binname="consul.exe"
-            fi
-            CGO_ENABLED=0 GOOS=${os} GOARCH=${arch} go install -ldflags "${GOLDFLAGS}" -tags "${GOTAGS}" && cp "${MAIN_GOPATH}/bin/${GOBIN_EXTRA}${binname}" "${outdir}/${binname}"
-            if test $? -ne 0
-            then
-               err "ERROR: Failed to build Consul for ${osarch}"
-               rm -r pkg.bin.new
-               return 1
-            fi
-         done
+         outdir="pkg.bin.new/${extra_dir}${os}_${arch}"
+         osarch="${os}/${arch}"
+         if test "${osarch}" == "darwin/arm" -o "${osarch}" == "darwin/arm64" -o "${osarch}" == "freebsd/arm64" -o "${osarch}" == "windows/arm" -o "${osarch}" == "windows/arm64" -o "${osarch}" == "freebsd/arm"
```

Comment:
This might be easier to scan if you used a `case...esac` structure instead of one huge or line.

Reply:
I went ahead and did that along with adding some comments.